### PR TITLE
Some fixes  in Get-Date -UFormat

### PR DESCRIPTION
**Pull request by iSazonov**
## PR Summary

In #4805 we fixed some formats for `-Uformat`.
Here we collect fixes of returned values for `-Uformat`.

1. Fix %s value
Address the [comment](https://github.com/PowerShell/PowerShell/issues/2195#issuecomment-375897359)
We use local datetime in evaluating %s format and UTC in epoch constant so returned value contains time zone shift compared to the value returned by Unix `date` utility.
In the PR we convert  local datetime to UTC.
https://stackoverflow.com/questions/2883576/how-do-you-convert-epoch-time-in-c

2. Fix %l uformat
Currently we have 0 .. 11 format.
Right format is 1..12 
See http://man7.org/linux/man-pages/man1/date.1.html

3. Fix %V uformat

PowerShell Committee conclusion https://github.com/PowerShell/PowerShell/pull/4508#issuecomment-325052526

Currently we use simple formula.
Right algorifm is described in https://en.wikipedia.org/wiki/ISO_week_date#Relation_with_the_Gregorian_calendar

Output format is changed from 1..53 to 01..53. See http://man7.org/linux/man-pages/man1/date.1.html

[Previous discussion](https://github.com/PowerShell/PowerShell/pull/4508/commits/f13a8b5d7e64ba5ea9c16c7e5408c5ca722f8bf7#r131528088).

.Net haven't ISO-compatible implementation.
I opened Issue in .Net Core repo https://github.com/dotnet/corefx/issues/28808 and get recomendation to use the workaround https://blogs.msdn.microsoft.com/shawnste/2006/01/24/iso-8601-week-of-year-format-in-microsoft-net/ with corrections based on comments.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by dantraMSFT** [comment]
@iSazonov Would you mind updating the PR description?  Your item #1 indicates your switching to UTC from local time but it doesn't indicate what problem is being fixed? (i.e., we do we need this change?)

Thanks,

**Comment by iSazonov** [comment]
@dantraMSFT Yes, I'll update - after the first commit I discovered there is still some breaking changes (to be compatible with Unix) and I want to bring them all together for consideration by PowerShell Committee.

**Comment by chuanjiao10** [comment]
Ok, I would like to help test night build if merged

**Comment by chuanjiao10** [comment]
6.1.0-preview.8834 ,tested, 
LGTM! thank you！
Who wants to fix %F, %N, %P, %W, %Z, %z？ Details at #2195

**Comment by iSazonov** [comment]
@chuanjiao10 I still don't look this cases. After the PR maybe.

**Comment by iSazonov** [comment]
Reopen the PR to restart CIs - last days sometimes they failed on restoring dotnet.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs [hunk 1]
```diff
@@ -393,7 +393,7 @@ private string UFormatDateString(DateTime dateTime)
                             break;
 
                         case 'l':
-                            sb.Append(StringUtil.Format("{0,2:0}", dateTime.Hour%12));
+                            sb.Append("{0:%h}");
```
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs]: Using http://man7.org/linux/man-pages/man1/date.1.html as reference, this should be space padded: `{0,2:%h}`

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs [hunk 2]
```diff
@@ -449,7 +449,34 @@ private string UFormatDateString(DateTime dateTime)
...
+                                case DayOfWeek.Monday:
+                                case DayOfWeek.Tuesday:
+                                case DayOfWeek.Wednesday:
+                                    normalizedDatetime = dateTime.AddDays(3);
+                                    break;
+
+                                case DayOfWeek.Friday:
+                                case DayOfWeek.Saturday:
+                                case DayOfWeek.Sunday:
+                                    normalizedDatetime = dateTime.AddDays(-3);
```
Comment (SteveL-MSFT) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs]: In the MSDN blog post, the "cheat" is to add 3 days for Mon-Wed, why are you subtracting 3 days for Fri-Sun?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs (hunk 1 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs]: Yes.
Fixed.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs (hunk 2 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs]: Below in the blog post comments (Keno) contain the fix.
Test values from Wikipedia article pass tests only with the fix.

Comment (SteveL-MSFT) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/GetDateCommand.cs]: Ok, got it.  Thanks.

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**